pandas/core/frame.py

```diff
@@ -6741,16 +6742,38 @@ def _gotitem(
         **_shared_doc_kwargs,
     )
     @Appender(_shared_docs["aggregate"])
-    def aggregate(self, func, axis=0, *args, **kwargs):
+    def aggregate(self, func=None, axis=0, *args, **kwargs):
         axis = self._get_axis_number(axis)
 
+        relabeling, func, columns, order = reconstruct_func(func, *args, **kwargs)
+
         result = None
         try:
             result, how = self._aggregate(func, axis=axis, *args, **kwargs)
         except TypeError:
             pass
         if result is None:
             return self.apply(func, axis=axis, args=args, **kwargs)
+
+        if relabeling:
```

Comment:
can you make another function to do this, e.g. relable results.

an alternative to having 2 functions, is actually to have `reconstruct_func` return a class which keeps the state

e.g. 

```
reconstructor = reconstruct_func(.....)

# the _aggregate / apply logic

return reconstructor.get_result(result)

Reply:
nice idea!

I have been thinking about it, and for now, i will go for the first option. Since in generic and frame, there is slightly different manipulation after `_aggregate / apply logic`, and not sure if `resample/rollin` will also be slightly different, so I will keep this in mind and once all nested renaming is done, I will try to come up with a clean PR to optimise it!